packages/router/src/history.ts

```diff
@@ -2,12 +2,16 @@
  * External dependencies
  */
 import { createBrowserHistory } from 'history';
+import type { BrowserHistory } from 'history';
 
 /**
  * WordPress dependencies
  */
 import { buildQueryString } from '@wordpress/url';
 
+export type EnhancedHistory = BrowserHistory & {
+	getLocationWithParams: () => Location;
+};
 const history = createBrowserHistory();
```

Comment:
I recommend sticking to interfaces when possible:

```suggestion
export interface EnhancedHistory extends BrowserHistory {
	getLocationWithParams: () => Location;
}

const history = createBrowserHistory();
```